dtfs-central-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js

```diff
@@ -1,22 +1,18 @@
 const db = require('../../../drivers/db-client');
+const { DB_COLLECTIONS } = require('../../../constants/dbCollections');
 
 const getUtilisationReports = async (req, res) => {
-  const { bankId } = req.params;
   try {
-    const utilisationReportsCollection = await db.getCollection('utilisation-reports');
-    const filteredAndSortedUtilisationReports = await utilisationReportsCollection.aggregate([
-      {
-        $match: {
-          bankId,
-        },
-      },
-      {
-        $sort: {
-          year: 1,
-          month: 1,
-        },
-      },
-    ]).toArray();
+    const { bankId } = req.params;
+
+    const utilisationReportsCollection = await db.getCollection(DB_COLLECTIONS.UTILISATION_REPORTS);
+    const filteredUtilisationReports = await utilisationReportsCollection
+      .find({ 'bank.id': { $eq: bankId } })
+      .toArray();
+
+    const filteredAndSortedUtilisationReports = filteredUtilisationReports
+      .sort((a, b) => a.year - b.year || a.month - b.month);
```

Comment:
I always find the native js `.sort` functions unintuitive, and this becomes more so when there is more that one attribute taken into account.

I think this is doing "sort by year ascending, then by month ascending"? If that's the case we could make use of lodash to make this a bit easier to understand. e.g.
```
const sortBy = require('lodash/sortBy');
...
const filteredAndSortedUtilisationReports = sortBy(filteredUtilisationReports, ['year', 'month']);
```
`sortBy` orders ascending by default. if we wanted to order anything descending we could instead use `orderBy` like this
```
I think this makes things much easier to understand, which hopefully would lead to less errors in future.
const orderBy = require('lodash/orderBy');
...
const filteredAndSortedUtilisationReports = orderBy(filteredUtilisationReports, ['year', 'month'], ['asc', 'desc']);
```

Reply:
Yep agreed that this is hard to read & understand :) here I am sorting year and month ascending but at some point I do sort them by descending year so will add orderBy in there